### ISSUE TITLE
Internal Python Language Server error

### ISSUE THREAD

**Issue by guangyu-shi** [spyder-ide/spyder#14463]
### What steps will reproduce the problem?

The issue comes out when running 'python bootstrap.py' as follows:

 (spyder-env) gary@gary-HP-Compaq-Pro-6300-SFF:~/src/spyder| python bootstrap.py
Executing Spyder from source checkout
*. Patched sys.path with /home/gary/src/spyder
*. Patched sys.path with /home/gary/src/spyder/external-deps/python-language-server
*. Patched sys.path with /home/gary/src/spyder/external-deps/spyder-kernels
*. Installing PyLS locally on /home/gary/src/spyder/external-deps/python-language-server/.installation-dir
*. PyQt5 is detected, selecting
*. Imported Spyder 4.3.0.dev0 - Revision 7fad6af, Branch: 4.x
    [Python 3.6.9 64bits, Qt 5.12.10, PyQt5 5.12.3 on Linux]
*. Running Spyder
Bootstrap completed in 00:00:00.6279

### Traceback
```python-traceback
  File "/home/gary/src/spyder-env/lib/python3.6/site-packages/pyls_jsonrpc/endpoint.py", line 113, in consume
    self._handle_request(message['id'], message['method'], message.get('params'))
  File "/home/gary/src/spyder-env/lib/python3.6/site-packages/pyls_jsonrpc/endpoint.py", line 182, in _handle_request
    handler_result = handler(params)
  File "/home/gary/src/spyder-env/lib/python3.6/site-packages/pyls_jsonrpc/dispatchers.py", line 23, in handler
    return method(**(params or {}))
  File "/home/gary/src/spyder/external-deps/python-language-server/pyls/python_ls.py", line 209, in m_initialize
    processId, _kwargs.get('capabilities', {}))
  File "/home/gary/src/spyder/external-deps/python-language-server/pyls/config/config.py", line 53, in __init__
    entry_point.load()
  File "/home/gary/src/spyder-env/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2446, in load
    self.require(*args, **kwargs)
  File "/home/gary/src/spyder-env/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2469, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
  File "/home/gary/src/spyder-env/lib/python3.6/site-packages/pkg_resources/__init__.py", line 775, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)

pkg_resources.VersionConflict: (python-language-server 0+untagged.1.g7fad6af.dirty (/home/gary/src/spyder/external-deps/python-language-server), Requirement.parse('python-language-server>=0.36.2')
```

## Versions

* Spyder version: 4.3.0.dev0 7fad6af
* Python version: 3.6.9
* Qt version: 5.12.10
* PyQt5 version: 5.12.3
* Operating System: Linux 4.15.0-128-generic

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  4.0.0 (OK)
cloudpickle >=0.5.0             :  1.6.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.1.0 (OK)
IPython >=7.6.0                 :  7.16.1 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
keyring >=17.0.0                :  21.5.0 (OK)
nbconvert >=4.0                 :  6.0.7 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
parso =0.7.0                    :  0.7.0 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.7.3 (OK)
pygments >=2.0                  :  2.7.3 (OK)
pylint >=1.0                    :  2.6.0 (OK)
pyls >=0.36.2;<1.0.0            :  0+untagged.1.g7fad6af.dirty (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.3.0             :  0.3.0 (OK)
qdarkstyle >=2.8                :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.0.2 (OK)
qtconsole >=5.0.1               :  5.0.1 (OK)
qtpy >=1.5.0                    :  1.9.0 (OK)
setuptools >=39.0.0             :  51.0.0 (OK)
sphinx >=0.6.6                  :  3.3.1 (OK)
spyder_kernels >=1.10.1;<1.11.0 :  1.11.0.dev0 (OK)
textdistance >=4.2.0            :  4.2.0 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3               :  1.0.2 (OK)
xdg >=0.26                      :  0.27 (OK)
zmq >=17                        :  20.0.0 (OK)

# Optional:
cython >=0.21                   :  None (NOK)
matplotlib >=2.0.0              :  None (NOK)
numpy >=1.7                     :  None (NOK)
pandas >=1.1.1                  :  None (NOK)
scipy >=0.17.0                  :  None (NOK)
sympy >=0.7.3                   :  None (NOK)
```

**Comment by guangyu-shi** [comment]
python3 -m pip show python-language-server
Name: python-language-server
Version: 0.36.2
Summary: Python Language Server for the Language Server Protocol
Home-page: https://github.com/palantir/python-language-server
Author: Palantir Technologies, Inc.
Author-email: None
License: UNKNOWN
Location: /home/gary/src/spyder-env/lib/python3.6/site-packages
Requires: ujson, jedi, python-jsonrpc-server, pluggy

So the version of python-language-server is 0.36.2, which meets 'python-language-server>=0.36.2'

**Comment by ccordoba12** [comment]
Hey @guangyu-shi, it seems you cloned our repo without pulling our tags. That's my best guess.

Our instructions on how to work with Spyder from `git clone` are described in our [Contributing guide](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md#setting-up-a-development-environment). Please note that we prefer conda over pip.

**Comment by guangyu-shi** [comment]
I also tried conda; but it got another issue as follows:
gary@gary-Aspire-M3910:~$ spyder 
Traceback (most recent call last):
  File "/home/gary/src/miniconda3/bin/spyder", line 11, in <module>
    sys.exit(main())
  File "/home/gary/src/miniconda3/lib/python3.7/site-packages/spyder/app/start.py", line 186, in main
    from spyder.app import mainwindow
  File "/home/gary/src/miniconda3/lib/python3.7/site-packages/spyder/app/mainwindow.py", line 78, in <module>
    from qtpy.compat import from_qvariant
  File "/home/gary/src/miniconda3/lib/python3.7/site-packages/qtpy/compat.py", line 15, in <module>
    from .QtWidgets import QFileDialog
  File "/home/gary/src/miniconda3/lib/python3.7/site-packages/qtpy/QtWidgets.py", line 22, in <module>
    from PyQt5.QtWidgets import *
ImportError: libGL.so.1: cannot open shared object file: No such file or directory
